captum/attr/_core/layer/layer_lrp.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+from ...._utils.common import _format_input
+from ...._utils.gradient import (
+    apply_gradient_requirements,
+    compute_gradients,
+    undo_gradient_requirements,
+)
+from ..._core.lrp import LRP
+from ..._utils.attribution import LayerAttribution
+
+
+class LayerLRP(LRP, LayerAttribution):
+    r"""
+    Layer-wise relevance propagation is based on a backward propagation
+    mechanism applied sequentially to all layers of the model. Here, the
+    model output score represents the initial relevance which is decomposed
+    into values for each neuron of the underlying layers. The decomposition
+    is defined by rules that are chosen for each layer, involving its weights
+    and activations. Details on the model can be found in the original paper
+    [https://doi.org/10.1371/journal.pone.0130140]. The implementation is
+    inspired by the tutorial of the same group
+    [https://doi.org/10.1016/j.dsp.2017.10.011] and the publication by
+    Ancona et al. [https://openreview.net/forum?id=Sy21R9JAW].
+    """
+
+    def __init__(self, model, layer):
+        """
+        Args:
+
+            model (callable): The forward function of the model or
+                        any modification of it. Custom rules for a given layer need to
+                        be defined as attribute
+                        `module.rule` and need to be of type PropagationRule.
+            layer (torch.nn.Module, None): Layer for which attributions are computed.
```

Comment:
@nanohanno, we have started to support lists of layers instead of just a single layer in some of the algorithms. I think that it would make sense to support here lists of layers as well. Here is an example:
https://github.com/pytorch/captum/blob/master/captum/attr/_core/layer/layer_activation.py#L32 

To be more consistent, I'd, actually, prefer to pass all layers in a list instead of having the convention of layer=None.
If we have that convention here, we have to adjust it in other algorithms as well and computing attributions for all layers isn't very common for other algorithms. Let me know what do you think.

Reply:
That sounds useful. I changed to lists instead of None, also in the test cases.